### ISSUE TITLE
`awkward` required but not installed on `pip install vector`

### ISSUE THREAD

**Issue by lukasheinrich** [scikit-hep/vector#75]
congrats on the release! 

I see the following on a fresh env - is this intended?
```
python -m venv _testenv
source _testenv/bin/activate
pip install vector
python -m vector
```

produces

```
Traceback (most recent call last):
  File "/Users/lukasheinrich/.pyenv/versions/3.7.2/lib/python3.7/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/Users/lukasheinrich/.pyenv/versions/3.7.2/lib/python3.7/runpy.py", line 142, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/Users/lukasheinrich/.pyenv/versions/3.7.2/lib/python3.7/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/Users/lukasheinrich/Code/pylhe/_testenv/lib/python3.7/site-packages/vector/__init__.py", line 8, in <module>
    from vector._backends.awkward_ import VectorAwkward
  File "/Users/lukasheinrich/Code/pylhe/_testenv/lib/python3.7/site-packages/vector/_backends/awkward_.py", line 35, in <module>
    import awkward as ak
ModuleNotFoundError: No module named 'awkward'
```

**Comment by henryiii** [comment]
Oooh, looks like a bug worthy of a patch release! Been hoping for one of those, as I worked on the CI release system and PyPI page, but not enough to warrant a patch release with no code changes.  @jpivarski I'm wrapping the import of VectorAwkward in a try/catch, does that seem like the best idea? - sadly, I guess this needs to affect `__all__` too.

**Comment by henryiii** [comment]
It would have been nicer to put all the Awkward stuff in a vector.awkward submodule, then disallow that import if awkward was not available. As it is, it's ugly statically if VectorAwkward is optionally available or if it is possibly None. Top-level optional components are messy.

**Comment by jpivarski** [comment]
All of the Awkward stuff _is_ in a `vector._backends.awkward_` submodule, and it was only intended to be imported if Awkward was available. As stated in my review of PR #76, that was a last-minute slip-up.

**Comment by henryiii** [comment]
Should be fixed in 0.8.1!
